quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+# Todo - Start Here:
+#  1) Run setup.bat to install necessary dependencies.
+#  2) Follow the To-Do comments for instructions, tips and reference material
+#  3) run the script
+
+# Todo (Yochai) cleanup the import section when done
+import io
+import os
+import time
+import typing
+
+from datetime import timedelta
+
+from azure.kusto.data.exceptions import KustoClientError, KustoServiceError
+from azure.kusto.data.helpers import dataframe_from_result_table
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder, ClientRequestProperties
+
+from azure.kusto.ingest import (
+    QueuedIngestClient,
+    IngestionProperties,
+    FileDescriptor,
+    BlobDescriptor,
+    StreamDescriptor,
+    DataFormat,
+    ReportLevel,
+    IngestionMappingType,
+    KustoStreamingIngestClient,
+)
+
+# Todo - Config (Auto-Filled properties):
+kustoUri = "https://yogiladadx.westeurope.dev.kusto.windows.net"
+ingestUri = "https://ingest-yogiladadx.westeurope.dev.kusto.windows.net"
+databaseName = "e2e"
+tableName = "SampleTable"
+tableSchema = "SampleTable_schema"
+# Todo - Learn More: For additional information about supported data formats, see
+#  https://docs.microsoft.com/en-us/azure/data-explorer/ingestion-supported-formats
+fileFormat = DataFormat.CSV
+
+# Todo - Config (Optional): Change the authentication method from Device Code (User Prompt) to one of the other options
+#  Some of the auth modes require additional environment variables to be set in order to work (check the use below)
+#  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+authenticationMode = "deviceCode"  # choose between: (deviceCode|managedIdentity|AppKey|AppCertificate)

```

Comment:
there is also interactive auth which works pretty well

Reply:
Awesome. Forgot about this